backend/testing/tester.py

```diff
@@ -0,0 +1,153 @@
+        if isinstance(body, dict):
+            event = body
+        else:
+            event = json.loads(body)
+        # logging.debug(event)
+
+        if message.delivery_info['routing_key'] == 'update-insert':
+            expected = messages[curr_idx]['detection_update_response']
+        elif message.delivery_info['routing_key'] == 'update':
+            expected = messages[curr_idx]['detection_hijack_response']
+        elif message.delivery_info['routing_key'] == 'hijack-update':
+            expected = messages[curr_idx]['database_hijack_response']
+
+        for key in set(event.keys()).intersection(expected.keys()):
+            if event[key] != expected[key]:
+                sys.exit('Unexpected value for key \"{}\"\nReceived: {}, Expected: {}'
+                         .format(key, event[key], expected[key]))
+
+        expected_messages -= 1
+        if expected_messages <= 0:
+            curr_idx += 1
+        message.ack()
+
+    def send_next_message(conn):
+        global curr_idx, expected_messages
+        with conn.Producer() as producer:
+            expected_messages = len(messages[curr_idx]) - 1
+            print('Publishing #{}'.format(curr_idx))
+            # logging.debug(messages[curr_idx]['send'])
+
+            producer.publish(
+                messages[curr_idx]['send'],
+                exchange=update_exchange,
+                routing_key='update',
+                serializer='json'
+            )
+
+    def waitExchange(exchange, channel):
+        flag = True
+        while flag:
+            try:
+                exchange.declare(passive=True, channel=channel)
+                flag = False
+            except Exception:
+                time.sleep(1)
+
+    with Connection(RABBITMQ_HOST) as connection:
+        print('Waiting for hijack exchange..')
+        waitExchange(hijack_exchange, connection.default_channel)
+        print('Waiting for update exchange..')
+        waitExchange(update_exchange, connection.default_channel)
+        print('Waiting for pg_amq exchange..')
+        waitExchange(pg_amq_bridge, connection.default_channel)
+        time.sleep(10)
+
+        while send_cnt < send_len:
+            curr_idx = send_cnt
+            send_next_message(connection)
+            send_cnt += 1
+            with nested(
+                    connection.Consumer(
+                        hijack_queue,
+                        callbacks=[validate_message],
+                        accept=['pickle']
+                    ),
+                    connection.Consumer(
+                        update_queue,
+                        callbacks=[validate_message],
+                    ),
+                    connection.Consumer(
+                        hijack_db_queue,
+                        callbacks=[validate_message]
+                    )
+
+            ):
+                while curr_idx != send_cnt:
```

Comment:
Recheck this logic, I think this remains always True and thus testing is eventually aborted after 4 hours. Unapproving PR for now.